### PR TITLE
RBrowser breadcrumbs

### PR DESCRIPTION
**Pull request by Falcort**
This PR add the breadcrumbs to the RBrowser, and allow to move the working directory

### CODE REVIEW

File: gui/browserv7/inc/ROOT/RBrowser.hxx [hunk 1]
```diff
@@ -98,6 +98,7 @@ protected:
    std::string GetFileIcon(TString &name);
    std::string ProcessBrowserRequest(const std::string &msg);
    std::string ProcessDblClick(const std::string &path, const std::string &drawingOptions);
+   std::string getCurrentWorkingDirectory();
```
Comment (linev) [inline gui/browserv7/inc/ROOT/RBrowser.hxx]: Please use first capital letter for methods in c++ - this is ROOT naming convention

File: gui/browserv7/src/RBrowser.cxx [hunk 2]
```diff
@@ -651,6 +651,17 @@ void ROOT::Experimental::RBrowser::SendInitMsg(unsigned connid)
    fWebWindow->Send(connid, msg);
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////////
+/// Return the current directory of ROOT
+
+std::string ROOT::Experimental::RBrowser::getCurrentWorkingDirectory() {
```
Comment (linev) [inline gui/browserv7/src/RBrowser.cxx]: New line after method name:
```
std::string ROOT::Experimental::RBrowser::getCurrentWorkingDirectory() 
{
```

File: ui5/browser/controller/Browser.controller.js [hunk 3]
```diff
@@ -586,6 +638,7 @@ sap.ui.define(['sap/ui/core/Component',
       
       /** process initial message, now it is list of existing canvases */
       processInitMsg: function(msg) {
+         this.websocket.Send('WORKDIR:'); // Update the breadcrumbs
```
Comment (linev) [inline ui5/browser/controller/Browser.controller.js]: Better use message "GETWORKDIR" here. Need correspondent change in c++

File: ui5/browser/controller/Browser.controller.js [hunk 4]
```diff
@@ -361,13 +362,61 @@ sap.ui.define(['sap/ui/core/Component',
...
+        let oLinks = oBreadcrumbs.getLinks();
+        let path = "/";
+        for (let i = 1; i<oLinks.length; i++) {
+          if (oLinks[i].sId === sId ) {
+            path += oLinks[i].getText();
+            break;
+          }
+          path += oLinks[i].getText() + "/";
+        }
+        return myThis.websocket.Send('SWITCHWORKDIR:' + path);
```
Comment (bellenot) [inline ui5/browser/controller/Browser.controller.js]: Maybe something like:
```suggestion
        return myThis.websocket.Send('CHDIR:' + path);
```
would be explicit enough

File: gui/browserv7/inc/ROOT/RBrowser.hxx (hunk 1 above)
Comment (Falcort) [reply gui/browserv7/inc/ROOT/RBrowser.hxx]: Fixed in : 6404a15

File: gui/browserv7/src/RBrowser.cxx (hunk 2 above)
Comment (Falcort) [reply gui/browserv7/src/RBrowser.cxx]: Fixed in : bb144ba

File: ui5/browser/controller/Browser.controller.js (hunk 3 above)
Comment (Falcort) [reply ui5/browser/controller/Browser.controller.js]: Fixed in : bb144ba

File: ui5/browser/controller/Browser.controller.js (hunk 4 above)
Comment (Falcort) [reply ui5/browser/controller/Browser.controller.js]: Fixed in : 7a42e78

**Comment by linev** [comment]
@Falcort Can you rebase your branch against current ROOT master and submit again - now there are conflicts and I cannot merge it without manual intervention

**Comment by linev** [comment]
> Is that okay, with you if rebase tomorrow ?

Yes, of course

**Comment by linev** [comment]
@Falcort 
Are you ready with your changes? If yes, please rebase against master and do `push --force`.
Most probably, tomorrow I will submit new RBrowsable code, which fully change server-side implementation of RBrowser

**Comment by Falcort** [comment]
@linev 
Everything is now working, and i am actually rebasing on master

**Comment by Falcort** [comment]
@linev or @bellenot you can remove the DO NOT MERGE label

**Comment by Falcort** [comment]
@linev I was able to fix all the problems, eveythings is working now
I have used a global var to indicate if the Browser reply need to change the TreeTable of just populate it
You can now change to any directory in your computers

**Comment by linev** [comment]
You just remove commit. One should do something like:

```
git fetch upstream
git rebase upsteram/master
... resolve conflicts ...
git push --force
```

**Comment by Falcort** [comment]
@linev Thanks, but it's already done ^^
I am actually testing my rebase, and will push soon

**Comment by Falcort** [comment]
@linev I think my rebase is ok now

**Comment by linev** [comment]
Finally!

**Comment by Falcort** [comment]
@linev Sorry, I did have some problems building root, and having troubles with the rebase

**Comment by linev** [comment]
@phsft-bot build with flags -DCMAKE_CXX_STANDARD=14 -Droot7=ON

**Comment by linev** [comment]
Just let jenkins check your changes.

**Pull request merged**